src/app/pages/component-sidenav/component-sidenav.ts

```diff
@@ -37,17 +43,86 @@ export class ComponentSidenav implements OnInit {
         this.sidenav.close();
       }
     });
+
     // Combine params from all of the path into a single object.
     this.params = Observable.combineLatest(
-      this._route.pathFromRoot.map(route => route.params),
-      Object.assign);
+        this._route.pathFromRoot.map(route => route.params),
+        Object.assign);
   }
 
   isScreenSmall(): boolean {
     return this.mediaMatcher.matches;
   }
 }
 
+@Component({
+  selector: 'app-component-nav',
+  templateUrl: './component-nav.html',
+  animations: [
+    trigger('bodyExpansion', [
+      state('collapsed', style({height: '0px', visibility: 'hidden'})),
+      state('expanded', style({height: '*', visibility: 'visible'})),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4,0.0,0.2,1)')),
+    ]),
+  ],
+})
+export class ComponentNav implements OnInit, OnDestroy {
+
+  @Input() params: Observable<Params>;
+  expansions = {};
+  private _onDestroy = new Subject<void>();
+
+  constructor(public docItems: DocumentationItems,
+              private _router: Router) { }
+
+  ngOnInit() {
+    this._router.events
+      .switchMap(() => this.params)
+      .takeUntil(this._onDestroy)
+      .subscribe(p => this.setExpansions());
+
+    this.setExpansions();
+  }
+
+  ngOnDestroy() {
+    this._onDestroy.next();
+    this._onDestroy.complete();
+  }
+
+  setExpansions() {
+    this.params.subscribe(p => {
+      const categories = this.docItems.getCategories(p.section);
+      for (const category of categories) {
```

Comment:
As a followup of a previous comment, this should be,

```ts


setExpansions(p: Params) {
  const categories = this.docItems.getCategories(p.section);
  for (...) {
    // ...
  }
}
```

so that it doesn't create another subscription. That also means that `ngOnInit` should updated like,

```ts
ngOnInit() {
  this._router.events
    .switchMap(() => this.params)
    .takeUntil(this._onDestroy)
    .subscribe(p => this.setExpansions(p));
}
```

If there is a problem that `_router.events` doesn't fire immediately, you could do this,

```ts
ngOnInit() {
  this._router.events
    .startWith(null)
    .switchMap(() => this.params)
    .takeUntil(this._onDestroy)
    .subscribe(p => this.setExpansions(p));
}
```

Reply:
Beautiful and done.